File: src/main/java/model/ModifiableSnake.java
package model;

import model.strategy.SnakeStrategy;

public class ModifiableSnake extends Snake {

    public ModifiableSnake(ModifiableArena arena, SnakeStrategy strategy, String name) {
        super(arena, strategy, name);
    }

    public void move() {
        Coordinate nextCoordinate = decideNextCoordinate();
        moveTo(nextCoordinate);
    }

}

File: snake-framework/src/main/kotlin/co/electric/snake/framework/model/ModifiableSnake.kt
package co.electric.snake.framework.model

import co.electric.snake.framework.strategy.SnakeStrategy

class ModifiableSnake(arena: ModifiableArena, strategy: SnakeStrategy, name: String) : Snake(arena, strategy, name) {

    fun move() {
        val nextCoordinate = decideNextCoordinate()
        moveTo(nextCoordinate)
    }

}
